datago/src/generator_http.rs

```diff
@@ -0,0 +1,403 @@
+
+    pub partitions_count: String,
+    pub partition: String,
+}
+
+// implement a helper to get the http request which correspons to the db request structure above
+impl DbRequest {
+    pub fn get_http_request(&self, api_url: &str, api_key: &str) -> reqwest::blocking::Request {
+        let mut url = if self.random_sampling {
+            Url::parse(&format!("{}images/random/", api_url)).unwrap()
+        } else {
+            Url::parse(&format!("{}images/", api_url)).unwrap()
+        };
+
+        let mut query_pairs = url.query_pairs_mut();
+
+        let maybe_add_field =
+            |query_pairs: &mut url::form_urlencoded::Serializer<url::UrlQuery>,
+             field: &str,
+             value: &str| {
+                if !value.is_empty() {
+                    query_pairs.append_pair(field, value);
+                }
+            };
+
+        let return_latents = if !self.has_latents.is_empty() {
+            format!("{},{}", self.has_latents, self.has_masks)
+        } else {
+            self.has_masks.clone()
+        };
+
+        maybe_add_field(&mut query_pairs, "fields", &self.fields);
+        maybe_add_field(&mut query_pairs, "sources", &self.sources);
+        maybe_add_field(&mut query_pairs, "sources__ne", &self.sources_ne);
+        maybe_add_field(&mut query_pairs, "page_size", &self.page_size);
+
+        maybe_add_field(&mut query_pairs, "tags", &self.tags);
+        maybe_add_field(&mut query_pairs, "tags__ne", &self.tags_ne);
+
+        maybe_add_field(&mut query_pairs, "has_attributes", &self.has_attributes);
+        maybe_add_field(&mut query_pairs, "lacks_attributes", &self.lacks_attributes);
+
+        maybe_add_field(&mut query_pairs, "has_masks", &self.has_masks);
+        maybe_add_field(&mut query_pairs, "lacks_masks", &self.lacks_masks);
+
+        maybe_add_field(&mut query_pairs, "has_latents", &self.has_latents);
+        maybe_add_field(&mut query_pairs, "lacks_latents", &self.lacks_latents);
+        maybe_add_field(&mut query_pairs, "return_latents", &return_latents);
+
+        maybe_add_field(&mut query_pairs, "short_edge__gte", &self.min_short_edge);
+        maybe_add_field(&mut query_pairs, "short_edge__lte", &self.max_short_edge);
+        maybe_add_field(&mut query_pairs, "pixel_count__gte", &self.min_pixel_count);
+        maybe_add_field(&mut query_pairs, "pixel_count__lte", &self.max_pixel_count);
+
+        maybe_add_field(&mut query_pairs, "duplicate_state", &self.duplicate_state);
+
+        maybe_add_field(&mut query_pairs, "partitions_count", &self.partitions_count);
+        maybe_add_field(&mut query_pairs, "partition", &self.partition);
+
+        drop(query_pairs);
+
+        let mut headers = HeaderMap::new();
+        headers.insert(
+            AUTHORIZATION,
+            HeaderValue::from_str(&format!("Token  {}", api_key)).unwrap(),
+        );
+
+        let mut req = reqwest::blocking::Request::new(reqwest::Method::GET, url);
+        req.headers_mut().extend(headers);
```

Comment:
```suggestion
        let mut req = reqwest::blocking::Request::new(reqwest::Method::GET, url);
        req.headers_mut().append(AUTHORIZATION, HeaderValue::from_str(&format!("Token {}", api_key)).unwrap());
```

Reply:
note: changing this to an expect, report an issue with the API key